### PR TITLE
feature/controls-in-config

### PR DESCRIPTION
**Pull request by tzinckgraf**
- moved layers into a layers directory, controls into a controls
directory
- controls created dynamically
- flyTo isn't a real control, so it is removed
- added more types
- added FIXMEs for all type issues

### CODE REVIEW

File: src/components/controls/Slider.svelte [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  onMount(() => {
+    // FIXME move this function somewhere else
+    const id = `${layerProperty.id}.${controlProperty.type}`;
+    store = GetOrCreateControlStore(id);
+    args = controlProperty.args as SliderProperties;
+    value = args.initialValue;
+  });
+
+  $: {
+    if (args !== undefined) {
```
Comment (tzinckgraf) [inline src/components/controls/Slider.svelte:25]: Things like this are until I learn more about svelte life-cycle. It's a work in progress. I can put a FIXME here.

File: src/components/layers/LayerByValueList.svelte [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+        Currently, the value is used to determine opacity of the feature outline
+    */
+
+    import { onMount, onDestroy } from 'svelte';
+    import { Writable } from "svelte/store"
+    import chroma from "chroma-js";
+    import L from 'leaflet';
+
+	import GeoJsonLayer from './GeoJsonLayer.svelte';
```
Comment (tzinckgraf) [inline src/components/layers/LayerByValueList.svelte:23]: Looks like a tab snuck in here. I will fix it.

File: src/components/layers/LayerByValueList.svelte [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            return `${args.prefixKey}|All`; 
+        }
+        if (keys.length == 0) return;
+        const filteredKeys = keys 
+            .map(x => parseInt(x.split('|')[1]))
+            .filter(x => x < value);
+        return `${args.prefixKey}|${filteredKeys[filteredKeys.length - 1]}`;
+    }
+
+
```
Comment (tzinckgraf) [inline src/components/layers/LayerByValueList.svelte:78]: FIXME two new lines.

**Comment by tzinckgraf** [comment]
Updated and tested.

File: src/components/layers/LayerControl.svelte [hunk 4]
```diff
@@ -1,32 +1,42 @@
...
-
-    import { layersStore } from '../stores.js';
+    import { layersStore } from '../../stores';
 
     export let config: Config|undefined = undefined;
-    let control = undefined;
-    let layers = [];
-    let layersMap = {};
+
+    // FIXME types, but need to figure out leaflet types here
```
Comment (njason) [inline src/components/layers/LayerControl.svelte:12]: What's wrong with the types that are there?

Comment (tzinckgraf) [reply src/components/layers/LayerControl.svelte:12]: I think I did actually fix this and thought I put it in another PR.

File: src/components/layers/LayerByValueList.svelte [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    let unsubscribe: (() => void);
+
+    // FIXME types
+    let colorChroma: any;
+
+    onMount(() => {
+        // FIXME hard-coded to connect to a slider.
+        const id = `${property.id}.slider`;
+        args = property.args;
+        store = GetOrCreateControlStore(id);
```
Comment (njason) [inline src/components/layers/LayerByValueList.svelte:45]: Why not pass args.initialValue in here?

Comment (tzinckgraf) [reply src/components/layers/LayerByValueList.svelte:45]: In the store? I am still figuring out life cycle. If two components call the store, who reads it first? I am not sure. What if the other place uses an initial value? I don't have a good answer for these yet, so I didn't include that fix in this PR.

One PR for one fix. This creates controls. It does not fix the store. That will require more work, more research, more testing, more code and more time.

**Pull request closed without merging**